packages/widgets/src/renderwidget.ts

```diff
@@ -0,0 +1,225 @@
+/*-----------------------------------------------------------------------------
+| Copyright (c) 2014-2017, PhosphorJS Contributors
+|
+| Distributed under the terms of the BSD 3-Clause License.
+|
+| The full license is in the file LICENSE, distributed with this software.
+|----------------------------------------------------------------------------*/
+import {
+  Message, MessageLoop
+} from '@phosphor/messaging';
+
+import {
+  ISignal
+} from '@phosphor/signaling';
+
+import {
+  VirtualContent, VirtualDOM
+} from '@phosphor/virtualdom';
+
+import {
+  Widget
+} from './widget';
+
+
+/**
+ * An object which can be used as a model for a render widget.
+ */
+export
+interface IRenderModel {
+  /**
+   * A signal emitted when the model state has changed.
+   *
+   * #### notes
+   * If this signal is provided, the render widget will automatically
+   * update whenever the signal is emitted.
+   */
+  readonly stateChanged?: ISignal<this, void>;
+}
+
+
+/**
+ * A widget which renders its content using the virtual DOM.
+ *
+ * #### Notes
+ * Most subclasses will typically only implement the abstract `render()`
+ * method. Advanced use cases may reimplement some of the other methods.
+ */
+export
+abstract class RenderWidget<T extends IRenderModel = {}> extends Widget {
```

Comment:
In JLab we have stopped using the `Widget` suffix in classnames as it is redundant. Fine here unless we have a better idea.